pkg/scalers/rabbitmq_scaler.go

```diff
@@ -100,24 +100,20 @@ var rabbitmqLog = logf.Log.WithName("rabbitmq_scaler")
 
 // NewRabbitMQScaler creates a new rabbitMQ scaler
 func NewRabbitMQScaler(config *ScalerConfig) (Scaler, error) {
+	s := &rabbitMQScaler{}
+
 	metricType, err := GetMetricTargetType(config)
 	if err != nil {
 		return nil, fmt.Errorf("error getting scaler metric type: %s", err)
 	}
+	s.metricType = metricType
 
 	meta, err := parseRabbitMQMetadata(config)
 	if err != nil {
 		return nil, fmt.Errorf("error parsing rabbitmq metadata: %s", err)
 	}
-	httpClient := kedautil.CreateHTTPClient(meta.timeout, false)
-
-	if meta.protocol == httpProtocol {
-		return &rabbitMQScaler{
-			metricType: metricType,
-			metadata:   meta,
-			httpClient: httpClient,
-		}, nil
-	}
+	s.metadata = meta
+	s.httpClient = kedautil.CreateHTTPClient(meta.timeout, false)
 
 	// Override vhost if requested.
```

Comment:
This is wrong. I accidentally removed the `if meta.protocol == httpProtocol` check.